modules/kafka/kafka-dest-worker.c

```diff
@@ -102,6 +102,89 @@ kafka_dest_worker_calculate_topic(KafkaDestWorker *self, LogMessage *msg)
   return kafka_dest_worker_get_literal_topic(self);
 }
 
+static gboolean
+_transaction_init(KafkaDestWorker *self)
+{
+#ifdef SYSLOG_NG_HAVE_RD_KAFKA_INIT_TRANSACTIONS
+  KafkaDestDriver *owner = (KafkaDestDriver *) self->super.owner;
+
+  if (!owner->transaction_commit)
+    return TRUE;
+
+  if (owner->transaction_inited)
+    return TRUE;
+
+  rd_kafka_error_t *error = rd_kafka_init_transactions(owner->kafka, -1);
```

Comment:
Just a point for improvement: 
I didn't consider the `timeout` parameters of `rd_kafka_init_transactions` and `rd_kafka_commit_transaction` as it seemed fine to block on these calls. However, during shutdown/reload this could take a long time (it's configurable though e.g. `transaction.timeout.ms` defaults to 60 seconds).

We have `flush_timeout_on_shutdown` and `flush_timeout_on_reload` options. We could reuse them even if the option names are tied to the `rd_kafka_flush` call.